e3sm_diags/driver/utils/dataset_xr.py

```diff
@@ -848,11 +858,25 @@ def _get_dataset_with_derivation_func(
         if func in FUNC_NEEDS_TARGET_VAR:
             func_args = [target_var_key] + func_args  # type: ignore # pragma: nocover
 
+        # If the target variable key contains a wildcard, there are can be
+        # N number of function arguments. For the cases, we need to pass the
+        # entire list of DataArrays to the derived function in order to
+        # properly maintain attributes (e.g., "units"). For example,
+        # "bc_a?_CLXF" uses the Python `sum()` function, which sums all of the
+        # DataArrays by iterating over them and add all elements by index (rather
+        # than a typical sum reduction across an axis). Using `.sum()` without
+        # using `with xr.set_options(keep_attrs=True)` will result in attributes
+        # being dropped, resulting in potential downstream issues such as unit
+        # conversions which require the "units" attribute.
+        if self.is_src_vars_wildcard:
+            derived_var = func(func_args)  # pragma: nocover
+        else:
+            derived_var = func(*func_args)  # pragma: nocover
+
```

Comment:
The section of code that uses `is_srcs_vars_wildcard` to determine how to pass `func_args` (function args, aka the variables).